code/modules/hydroponics/grown.dm

```diff
@@ -111,30 +111,31 @@
 
 /// Turns the nutriments and vitamins into the distill reagent or fruit wine
 /obj/item/food/grown/proc/ferment()
+	var/reagent_purity = seed.get_reagent_purity()
+	var/purity_above_base = clamp((reagent_purity - 0.5) * 2, 0, 1)
+	var/quality_min = 0
+	var/quality_max = DRINK_FANTASTIC
+	var/quality = round(LERP(quality_min, quality_max, purity_above_base))
 	for(var/datum/reagent/reagent in reagents.reagent_list)
-		if(reagent.type != /datum/reagent/consumable/nutriment && reagent.type != /datum/reagent/consumable/nutriment/vitamin)
+		if(!istype(reagent, /datum/reagent/consumable))
```

Comment:
Is there a reason why this is changing?

Reply:
I want to make biogen work with any consumable, respecting nutrition values, instead of just converting 1 nutriment or 1 vitamin to 1 biomass. Not in code yet.

Reply:
This'll cause a lot more than sugar to be fermented, like other alcohol, so I'm not sure it's super what we want. Maybe it should be a typecache?

Reply:
Well this is a ferment() proc that is called when you put fruit into a barrel or it is fermented in the barrel-melon. I believe it is it's fine that other alcohol reagents will be turned into the other. I actually can't think of real case when this should happen unless you inject fruit with some other alcohol on purpose.